hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientRingbufferProxy.java

```diff
@@ -208,7 +210,16 @@ public E readOne(long sequence) throws InterruptedException {
         checkSequence(startSequence);
         checkNotNegative(minCount, "minCount can't be smaller than 0");
         checkTrue(maxCount >= minCount, "maxCount should be equal or larger than minCount");
-        checkTrue(minCount <= capacity(), "the minCount should be smaller than or equal to the capacity");
+
+        try {
+            checkTrue(minCount <= capacity(), "the minCount should be smaller than or equal to the capacity");
```

Comment:
according to the interface, this should throw `IllegalArgumentException` when `minCount` is larger than `capacity`. Should handle two cases differently:
- `capacity()` throwing exception
- `minCount` being larger than capacity